draft-ietf-dnsop-structured-dns-error.md

```diff
@@ -402,7 +399,12 @@ JSON:
   EDE option that conveys the information in the "c", "s", and "j"
   fields encoded in the JSON object.
 
+* If the local application that triggered the DNS request has more specific
+  context (e.g., contact information) that the DNS server, the content of
```

Comment:
```suggestion
  context (e.g., contact information) than the DNS server, the content of
```

Reply:
I wonder how the application can have more context than the DNS server itself ! 
My understanding is in scenarios where the application is pre-configured with the DNS server (e..g, TRR in browsers), it may have the contact information as part of the prior business agreement.

Reply:
This may not be possible to all error type. For regulatory filtering for example, the app may have some context.

Reply:
In case of regulatory filtering, error code will be set to "Censored" and the sub-error codes defined in the draft are not applicable. Even in this case, I don't get how the app would have more context about the contact information. 

Why wouldn't the DNS server know the contact details in case of regulatory filtering ?

I would stick to the generic use case (deployed today by all major browsers) of pre-configured resolvers that may have contact information. I propose the following update:

"If the DNS client has more specific context (e.g., contact information) about the DNS server...".

Reply:
Not sure to get your comment. The contact is not about the dns server, but whether an app can override for whatever reason the c reported by the DNS server.

Reply:
Please take an example to explain how the app would have specific context than the DNS server to override the c reported by the DNS server.

Reply:
An app may decide to redirect all claim requests to a single contact point. The app client can supply the contact returned by the DNS server, if needed.

I don't think we can mandate that if a contact is established, it MUST be with the info in the "c".